File: backend/AuthenticationSystem/src/main/java/com/vbt/AuthenticationSystem/auth/dto/TokenResponse.java
package com.vbt.AuthenticationSystem.auth.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.vbt.AuthenticationSystem.user.dto.UserResponse;

@JsonInclude(JsonInclude.Include.NON_NULL) // User null ise JSON'da gözükmeyecek
public record TokenResponse(
        @JsonProperty("access_token")
        String accessToken,

        @JsonProperty("refresh_token")
        String refreshToken,

        @JsonProperty("token_type")
        String tokenType,

        @JsonProperty("expires_in")
        long expiresIn,

        @JsonProperty("user")
        UserResponse user
) {
}

File: mobile/lib/features/authentication/data/models/token_response.dart
import 'user_model.dart';

class TokenResponse {
  final String accessToken;
  final String refreshToken;
  final String tokenType;
  final int expiresIn;
  final UserModel? user;

  const TokenResponse({
    required this.accessToken,
    required this.refreshToken,
    required this.tokenType,
    required this.expiresIn,
    this.user,
  });

  factory TokenResponse.fromJson(Map<String, dynamic> json) {
    return TokenResponse(
      accessToken: json["access_token"],
      refreshToken: json["refresh_token"],
      tokenType: json["token_type"],
      expiresIn: json["expires_in"],
      user: json["user"] == null
          ? null
          : UserModel.fromJson(json["user"]),
    );
  }
}